docs/admin/kubeadm.md

```diff
@@ -108,13 +109,15 @@ By default, `kubeadm init` does not set node CIDR's for pods and allows you to
 bring your own networking configuration through a CNI compatible network
 controller addon such as [Weave Net](https://github.com/weaveworks/weave-kube),
 [Calico](https://github.com/projectcalico/calico-containers/tree/master/docs/cni/kubernetes/manifests/kubeadm)
-or [Canal](https://github.com/tigera/canal/tree/master/k8s-install/kubeadm).
-If you are using a compatible cloud provider or flannel, you can specify a
-subnet to use for each pod on the cluster with the `--pod-network-cidr` flag.
+or [Canal](https://github.com/tigera/canal/tree/master/k8s-install/kubeadm), solutions
```

Comment:
This whole passage still feels very awkward to me.  how about something like:

> For certain networking solutions the Kubernetes master can also play a role in allocating network ranges (CIDRs) to each node.  This includes many cloud providers (link?) and flannel.  You can specify a subnet range that will be broken down and handed out to each node with the `--pod-network-cidr` flag.  This should be a minimum of a /16 so that `kubeadm` is able to assign /24 subnets to each node in the cluster. If you are using flannel with the manifest linked above you should use `--pod-network-cidr=10.244.0.0/16`.  Most CNI based networking solutions do not require this flag (this includes...)